### PR TITLE
New pop-ups, search for address, add all datasets

### PR DESCRIPTION
**Pull request by paylakatel**
This PR does a lot! Apologies for the size - others will be smaller.
We are:
- adding all datasets related to capital projects
- adding the ability to search for an address
- building out pop-ups as a table at the bottom of the map container
- updating the legend for the new datasets
- doing some small design updates

### CODE REVIEW

File: components/Layout.js [hunk 1]
```diff
@@ -71,20 +72,66 @@ export default class Layout extends React.Component {
             border: 3px solid #091f2f;
             font-family: 'Lora', serif;
             font-style: italic;
+            border-radius: 0px;
+          }
+
+          // geocoder style
+          .mapboxgl-ctrl-geocoder input {
+            background-image: url(https://www.boston.gov/crispus/images/public/icons/search.svg);
```
Comment (fionawhim) [inline components/Layout.js]: Can you use `https://patterns.boston.gov/images/public/icons/search.svg` instead?

File: components/Map.js [hunk 2]
```diff
@@ -59,7 +77,80 @@ class Map extends React.Component {
...
 
+    // In order to add a geocoder to our map, we need a mapbox access token.
+    // We've stored that using environment variables inside out nextjs config file.
+    const { publicRuntimeConfig } = getConfig();
+    const accessToken = publicRuntimeConfig.MapboxAccessToken;
+
+    const geocoder = new MapboxGeocoder({
+      accessToken: accessToken,
+      flyTo: true,
+      placeholder: 'Show projects near...',
```
Comment (fionawhim) [inline components/Map.js]: Ellipsis character: …

File: components/Map.js [hunk 3]
```diff
@@ -59,7 +77,80 @@ class Map extends React.Component {
...
+      // sit below the other layers.
+      // As a result, we load the geocoding point first, as we want that
+      // to sit under the other layers to keep it from interfering with click
+      // events on the other layers.
+      // Next is City Council Districts as that is a contextual layer. The
+      // public works street projects, public works ramps, budget
+      // facilities, and highlight layers are loaded next in that order.
+
+      // We add an empty geojson source and layer that we'll populate
+      // with the results of the geocoding search when appropriate.
```
Comment (fionawhim) [inline components/Map.js:129]: good comment

File: components/Table.js [hunk 4]
```diff
@@ -0,0 +1,138 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default function Table(props) {
```
Comment (fionawhim) [inline components/Table.js:4]: Good use of a separate component

**Review by fionawhim — APPROVED** [review]

File: components/Layout.js (hunk 1 above)
Comment (paylakatel) [reply components/Layout.js]: 👍

File: components/Map.js (hunk 2 above)
Comment (paylakatel) [reply components/Map.js]: Didn't know that was a thing! Updated it.

**Pull request merged**